tasks/connectors/cobaltio/cobaltio.rb

```diff
@@ -0,0 +1,171 @@
+              default: false,
+              description: "strip colon and following data from packageManager - used in asset file locator" },
+            { name: "package_strip_colon",
+              type: "boolean",
+              required: false,
+              default: false,
+              description: "strip colon and following data from package - used in asset file locator" },
+            { name: "kenna_connector_id",
+              type: "integer",
+              required: false,
+              default: nil,
+              description: "If set, we'll try to upload to this connector" },
+            { name: "kenna_api_key",
+              type: "api_key",
+              required: false,
+              default: nil,
+              description: "Kenna API Key" },
+            { name: "kenna_api_host",
+              type: "hostname",
+              required: false,
+              default: "api.kennasecurity.com",
+              description: "Kenna API Hostname" },
+            { name: "output_directory",
+              type: "filename",
+              required: false,
+              default: "output/cobaltio",
+              description: "If set, will write a file upon completion. Path is relative to #{$basedir}" }
+
+          ]
+        }
+      end
+
+      def run(opts)
+        super # opts -> @options
+
+        cobalt_api_token = @options[:cobalt_api_token]
+        cobalt_org_token = @options[:cobalt_org_token]
+
+        @kenna_api_host = @options[:kenna_api_host]
+        @kenna_api_key = @options[:kenna_api_key]
+        @kenna_connector_id = @options[:kenna_connector_id]
+
+        # output_directory = @options[:output_directory]
+
+        projectName_strip_colon = @options[:projectName_strip_colon]
+        packageManager_strip_colon = @options[:packageManager_strip_colon]
+        package_strip_colon = @options[:package_strip_colon]
+        to_date = Date.today.strftime("%Y-%m-%d")
+        retrieve_from = @options[:retrieve_from]
+        from_date = (Date.today - retrieve_from.to_i).strftime("%Y-%m-%d")
+
+        findings_json = cobalt_get_findings(cobalt_api_token, cobalt_org_token)
+        print_debug "findings json = #{findings_json}"
+
+        severity_map = { "high" => 7, "medium" => 5, "low" => 3 } # converter
+        state_map = { "need_fix" => "new", "wont_fix" => "risk_accepted", "valid_fix" => "resolved", "check_fix" => "in_progress", "carried_over" => "new" }
+        findings_json.each do |finding_obj|
+          next if cobalt_exclude_finding(finding_obj)
+
+          resource = finding_obj["resource"]
+          links = finding_obj["links"]
+          log = resource["log"]
+
+          asset_id = resource["asset_id"]
+          cobalt_asset = cobalt_get_asset(cobalt_api_token, cobalt_org_token, asset_id)
+          application = cobalt_asset["resource"]["title"] if cobalt_asset
+
+          asset = {
+            "external_id" => asset_id,
+            "application" => application,
```

Comment:
right now the application appears to be the same as the locator see my comments above. Are there different pages(urls) that make up "Saxophone external network"?

Reply:
Yes, typically an asset will have more than one page, see also my comment above.